frontend/test/currentUser/currentUser.store.spec.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { CurrentUserActions } from '@/v5/store/currentUser/currentUser.redux';
+import {
+	selectPersonalDataIsUpdating,
+	selectApiKeyIsUpdating,
+	selectCurrentUser,
+	selectFirstName,
+} from '@/v5/store/currentUser/currentUser.selectors';
+import { currentUserMockFactory, generatePersonlData } from './currentUser.fixtures';
+import { createTestStore } from '../test.helpers';
+
+describe('CurrentUser: store', () => {
+	let dispatch, getState;
+
+	beforeEach(() => {
+		({ dispatch, getState} = createTestStore());
+	})
+
+	it('should set currentUser', () => {
+		const mockCurrentUser = currentUserMockFactory();
+		dispatch(CurrentUserActions.fetchUserSuccess(mockCurrentUser));
+		const currentUser = selectCurrentUser(getState());
+		expect(currentUser).toEqual(mockCurrentUser);
+	})
+
+	describe('Updating currentUser attributes:', () => {
+		it('should update personal data', () => {
+			const mockCurrentUser = currentUserMockFactory();
+			dispatch(CurrentUserActions.fetchUserSuccess(mockCurrentUser));
+			const { firstName: mockFirstName } = generatePersonlData();
+			dispatch(CurrentUserActions.updateUserSuccess({ firstName: mockFirstName }));
+			const firstName = selectFirstName(getState());
```

Comment:
instead of using `selectFirstName`  can we use `selectCurrentUser(getState())` and test it against `{...mockCurrentUser, firstName:mockFirstName}` so you can test that `updateUserSuccess` doesnt breake anything else